### PR TITLE
Init sync refactor state initialization check + tests

### PR DESCRIPTION
**Pull request by farazdagi**
**What type of PR is this?**

> Other / Better Tests

**What does this PR do? Why is it needed?**
- This PR is a part of effort to improve test coverage of `init-sync` package. Our `service.go` file has a petty `~10%` of coverage - which is due to fact that it is not easy to test the service as it is. So, this file needs some, albeit minor, refactoring. The scope of this PR is simple: only code concerning with checking that beacon node head is ready (that code is extracted into its own method and well tested).

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**
- Previously we had to calls to `Unsubscribe()`, which was pointed out during audit, and which was an intended semantics and we added, a rather long, comment explaining it (see https://github.com/prysmaticlabs/prysm/pull/6376 for details): https://github.com/prysmaticlabs/prysm/blob/b928e9531c6b5b7ad4b8548404fd3e0867966d56/beacon-chain/sync/initial-sync/service.go#L85-L92
- As a part of this PR, we are moving the functionality out of `Start()` method (for testability), and as a part of such a refactoring, removing double call to `Unsubscribe()` (~~`break label` is used instead~~ it was possible to do it w/o label).
- cc @nisdas

### CODE REVIEW

**Review by Bit24888 — APPROVED** [review]

File: beacon-chain/sync/initial-sync/service.go [hunk 1]
```diff
@@ -239,3 +198,43 @@ func (s *Service) waitForMinimumPeers() {
...
+func (s *Service) waitForStateInitialization() time.Time {
+	var genesis time.Time
+	headState, err := s.chain.HeadState(s.ctx)
+	if headState == nil || err != nil {
+		log.Info("Waiting for state to be initialized")
+		// Wait for state to be initialized.
+		stateChannel := make(chan *feed.Event, 1)
+		stateSub := s.stateNotifier.StateFeed().Subscribe(stateChannel)
+		genesisSet := false
+	loop:
```
Comment (nisdas) [inline beacon-chain/sync/initial-sync/service.go:213]: we dont really use control statements like this in the repo, I dont see any advantage/necessity having `loop:` here.

File: beacon-chain/sync/initial-sync/service.go [hunk 2]
```diff
@@ -239,3 +201,38 @@ func (s *Service) waitForMinimumPeers() {
 		time.Sleep(handshakePollingInterval)
 	}
 }
+
+// waitForStateInitialization makes sure that beacon node is ready to be accessed: it is either
+// already properly configured or system waits up until state initialized event is triggered.
+func (s *Service) waitForStateInitialization() (time.Time, error) {
+	headState, err := s.chain.HeadState(s.ctx)
+	if err == nil && headState != nil {
```
Comment (nisdas) [inline beacon-chain/sync/initial-sync/service.go:209]: you need to handle the case where `err != nil`  also

Comment (farazdagi) [reply beacon-chain/sync/initial-sync/service.go:209]: How should I handle such a case? Originally, if error wasn't nil, the case was ignored and wait for event starts. This refactored implementation mimics exactly this behaviour. 

Do you want to halt on non-nil error and return from the function with that error (in which case `Start()` will also be halted)?

Comment (farazdagi) [reply beacon-chain/sync/initial-sync/service.go]: Error is explicitly returned now.

**Review by nisdas — APPROVED** [review]

**Pull request merged**